src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.tsx

```diff
@@ -1,95 +1,153 @@
-        <Card>
-          <CardBody>No data for {reading.data_type.long_name} in the last day.</CardBody>
-        </Card>
-      </Col>
+interface LoadDataCardProps {
+  platform: PlatformFeature
+  timeSeries: PlatformTimeSeries
+}
+
+/**
+ * Load the data for a time series and connect units to the data card to display
+ * additionally filter data to the last day
+ *
+ * @param platform
+ * @param timeSeries Time series to load data for
+ */
+const LoadDataCard: React.FC<LoadDataCardProps> = ({ platform, timeSeries }) => {
+  const { isLoading, data } = useDataset(timeSeries)
+  const unit_system = useUnitSystem()
+
+  if (isLoading) {
+    return <LoadingDataCard platform={platform} timeSeries={timeSeries} />
+  }
+
+  if (data) {
+    const aDayAgo = new Date()
+    aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+    const readings: ReadingTimeSeries[] = (data as DataTimeSeries).timeSeries.filter(
+      (reading) => aDayAgo < reading.time
     )
+
+    if (readings.length > 0) {
+      return (
+        <DataCardDisplay readings={readings} unit_system={unit_system} timeSeries={timeSeries} platform={platform} />
+      )
+    }
   }
 
-  const latest = data[data.length - 1]
-  const bounds = naturalBounds(reading.data_type.standard_name)
+  return <ErrorDataCard platform={platform} timeSeries={timeSeries} />
+}
 
-  const data_converter = converter(reading.data_type.standard_name)
+interface DataCardDisplayProps {
+  readings: ReadingTimeSeries[]
+  unit_system: UnitSystem
+  timeSeries: PlatformTimeSeries
+  platform: PlatformFeature
+}
+
+/**
+ * Format and display a data card
+ */
+export const DataCardDisplay: React.FC<DataCardDisplayProps> = ({ platform, readings, timeSeries, unit_system }) => {
+  const url = cardUrl(platform, timeSeries)
+  const data_converter = converter(timeSeries.data_type.standard_name)
+  const bounds = naturalBounds(timeSeries.data_type.standard_name)
 
-  data = data.map((r) => ({
+  const latest = readings[readings.length - 1]
+
+  const data = readings.map((r) => ({
     ...r,
     reading: round(data_converter.convertToNumber(r.reading, unit_system), 2),
   }))
 
   return (
-    <Col key={reading.data_type.standard_name} {...cardProps}>
-      <Link
-        to={urlPartReplacer(
-          urlPartReplacer(paths.platforms.observations, ":id", platform.id as string),
-          ":type",
-          reading.data_type.standard_name
-        )}
-      >
+    <Col {...cardProps}>
+      <Link to={url}>
         <Card>
           <CardHeader>
-            {reading.data_type.long_name} - {round(data_converter.convertToNumber(latest.reading, unit_system), 1)}{" "}
-            {data_converter.displayName(unit_system)} {convertUnit(reading.data_type.units, latest.reading)}
+            {timeSeries.data_type.long_name} - {round(data_converter.convertToNumber(latest.reading, unit_system), 1)}{" "}
+            {data_converter.displayName(unit_system)} {convertUnit(timeSeries.data_type.units, latest.reading)}
```

Comment:
Codacy found an issue: [Identifier 'data_converter' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)